src/datasets/utils/file_utils.py

```diff
@@ -466,6 +474,10 @@ def get_from_cache(
 
     # We don't have the file locally or we need an eTag
     if not local_files_only:
+        if url.startswith("ftp://"):
+            with closing(urllib.request.urlopen(url)) as r:
```

Comment:
Would it be possible to raise a `ConnectionError` here if the `urlopen` fails ?
Currently a bad ftp url raises raises an `URLError`

Can you also add a timeout ? (ex: 2secs)

Reply:
Perfect. I extracted this as a function (like `ftp_get`).